app/models/purchase.rb

```diff
@@ -1,13 +1,13 @@
 class Purchase < ApplicationRecord
-  belongs_to :user
   belongs_to :subcategory
+  belongs_to :purchaser, class_name: 'User', foreign_key: 'purchaser_id'
+  belongs_to :purchasee, class_name: 'User', foreign_key: 'purchasee_id'
 
```

Comment:
I wonder if `accepts_nested_attributes_for` might be useful here? (Disclaimer, I haven't used it myself).
* [Rails docs](https://api.rubyonrails.org/classes/ActiveRecord/NestedAttributes/ClassMethods.html)
* [Article](https://medium.com/@onyoo/why-accepts-nested-attributes-for-6ed190def58a)